projects/storefrontlib/src/lib/cms/store/actions/component.action.ts

```diff
@@ -18,12 +19,12 @@ export class LoadComponentFail implements Action {
 
 export class LoadComponentSuccess implements Action {
   readonly type = LOAD_COMPONENT_SUCCESS;
-  constructor(public payload: any) {}
+  constructor(public payload: Component) {}
```

Comment:
`<T extends Component>`. Check my other comments.

Reply:
This causes a syntax error

Reply:
```
loadComponent<T extends Component>(
    id: string,
    pageContext: PageContext,
    fields?: string
  ): Observable<T> {
```
causes errors
Type Object is not assignable to type T